taskweaver/memory/plugin.py

```diff
@@ -156,6 +158,38 @@ def to_dict(self):
             "enabled": self.enabled,
         }
 
+    def format_function_calling(self) -> Dict:
+        assert self.plugin_only is True, "Only `plugin_only` plugins can be called in this way."
+
+        def map_type(t: str) -> str:
+            if t.lower() == "string" or t.lower() == "str" or t.lower() == "text":
+                return "string"
+            if t.lower() == "integer" or t.lower() == "int":
+                return "integer"
+            if t.lower() == "float" or t.lower() == "double" or t.lower() == "number":
+                return "number"
+            if t.lower() == "boolean" or t.lower() == "bool":
+                return "boolean"
+            if t.lower() == "null" or t.lower() == "none":
+                return "null"
+            raise Exception(f"unknown type {t}")
```

Comment:
Why need to do such conversation for parameter type?

Reply:
to tolerate variations of types...

Reply:
Wondering which language does the unified types intent to comply? JSON or JavaScript/TypeScript or Python data types?

Reply:
https://json-schema.org/understanding-json-schema/reference/type